### ISSUE TITLE
[Question] How to Run 3D Version

### ISSUE THREAD

**Issue by rydeveraumn** [UW-ERSL/TOuNN#2]
Hey @aadityacs! This is some really cool work you have done here. Is there a way to run the 3D version that was explored in your paper? Thank you for the help.

**Comment by aadityacs** [comment]
Hi @rydeveraumn ,

Unfortunately the 3D solver happens to be under private wraps. Sorry about that

**Comment by rydeveraumn** [comment]
Hey @aadityacs! That makes sense to me! I was just hoping I wasn't overlooking something. Thank you for the info. Would you be able to link me to some resources about how to tackle the 3D FEA? If you have time of course if not no worries.

**Comment by mohdibntarek** [comment]
Shameless self promotion: TopOpt.jl (https://github.com/JuliaTopOpt/TopOpt.jl) allows you to implement the TOuNN approach quite easily using any NN architecture from https://github.com/FluxML/Flux.jl. See an example doing something similar in https://github.com/JuliaTopOpt/TopOpt.jl/blob/master/test/examples/neural.jl. TopOpt.jl supports 2D and 3D domains.

**Comment by rydeveraumn** [comment]
Nice! I will take shameless self promotion as long as I can learn some cool new things! Thanks for the resources! Julia too that is nice! I've heard there are some pretty nice speed ups in comparison with Python.

**Comment by mohdibntarek** [comment]
Feel free to join our #topopt channel on the Julia slack https://julialang.org/slack/ and say hi. We meet every 2 weeks to discuss anything topopt in Julia, open invitation.

**Comment by mohdibntarek** [comment]
Sorry @aadityacs for hijacking this issue.

**Comment by aadityacs** [comment]
No worries @mohamed82008 i might join the slack and say hello to! Glad to learn from the discussion there

And @rydeveraumn if you are considering starting out do consider Julia too! Imo it's definitely  friendlier too and you don't have to bother vectorizing everything like in python

**Comment by mohdibntarek** [comment]
Please do!

**Comment by rydeveraumn** [comment]
Hey @aadityacs! Nice to chat again :) I had a question do you have any repositories/work on thermal topology optimization?

**Comment by aadityacs** [comment]
Late reply @rydeveraumn  but you can find some implementation here https://github.com/UW-ERSL/AuTO

**Comment by nvnkrus** [comment]
Dear @aadityacs,

It is nice to meet you here. I have been studying your paper on "TOuNN: Topology Optimization using Neural Networks" and I have a question regarding the update of network parameters in the TOuNN method.

It seems that the sensitivity analysis for the loss function with respect to network parameters (Eq. 9 in the paper) is not explicitly used in the code, and the network parameters are updated only through backpropagation. I would appreciate it if you could clarify whether the sensitivity analysis for the loss function with respect to network parameters is indeed not utilized in the TOuNN method, and if the network parameters are updated solely through backpropagation.

Thank you for your time and assistance.
